File: src/345.reverse-vowels-of-a-string/345.reverse-vowels-of-a-string.cs
/*
 * @lc app=leetcode id=345 lang=csharp
 *
 * [345] Reverse Vowels of a String
 */
// Write a function that takes a string as input and reverse only the vowels of a string.

// Example 1:

// Input: "hello"
// Output: "holle"
// Example 2:

// Input: "leetcode"
// Output: "leotcede"
// Note:
// The vowels does not include the letter "y".


// @lc code=start
public class Solution
{
    public string ReverseVowels(string s)
    {
        // Accepted
        // 481/481 cases passed (144 ms)
        // Your runtime beats 5.9 % of csharp submissions
        // Your memory usage beats 25 % of csharp submissions (27.7 MB)
        char[] vowels = "AEIOUaeiou".ToCharArray(), schars = s.ToCharArray();
        int i = 0, l = s.Length;
        List<char> chars = new List<char> { };
        foreach (char c in schars)
        {
            if (vowels.Contains(c))
            {
                chars.Add(c);
            }
        }
        int chars_length = chars.Count() - 1;
        if (chars.Count() > 0)
        {
            while (i < l)
            {
                if (vowels.Contains(schars[i]))
                {
                    schars[i] = chars[chars_length];
                    chars_length --;
                }
                i++;
            }
        }
        return string.Join("", schars);
    }
}
// @lc code=end

File: src/345.reverse-vowels-of-a-string/345.reverse-vowels-of-a-string.py
#
# @lc app=leetcode id=345 lang=python3
#
# [345] Reverse Vowels of a String
#

# @lc code=start


class Solution:
    def reverseVowels(self, s: str) -> str:
        # Accepted
        # 481/481 cases passed (68 ms)
        # Your runtime beats 43.67 % of python3 submissions
        # Your memory usage beats 6.67 % of python3 submissions (15 MB)
        vowwels, n, l, chars = ("a", "e", "i", "o", "u",
                                "A", "E", "I", "O", "U"), 0, len(s), []
        for i in s:
            if i in vowwels:
                chars += [i]
        s = list(s)
        if chars:
            while n < l:
                if s[n] in vowwels:
                    s[n] = chars.pop()
                n += 1
        return "".join(s)
# @lc code=end
